### PR TITLE
fix(sumologicexporter): don't send null logs

### PR DESCRIPTION
**Pull request by pmalek-sumo**
To prevent sending `null` logs check if the body is not empty before attaching it like in the attached screenshot:

image

### CODE REVIEW

**Review by kkujawa-sumo — COMMENTED** [review]
Can the test case with sending null logs be added to unit tests? If yes, I recommend to add this.

**Comment by pmalek-sumo** [comment]
> Can the test case with sending null logs be added to unit tests? If yes, I recommend to add this.

I wanted to do this but it seems that it's non trivial to make the exporter send the empty logs due to logic in `appendAndSend`, i.e. the provided body there is empty (which is ok) and we cannot really configure `MaxRequestBodySize` to force the flush (in order to check the asserts in tests). 

I'm open to ideas to actually add a test for that.

ref: https://github.com/SumoLogic/sumologic-otel-collector/blob/2a5baaa9d851bf87738c59859c518b6825815490/pkg/exporter/sumologicexporter/sender.go#L400-L438

File: pkg/exporter/sumologicexporter/sender.go [hunk 1]
```diff
@@ -181,7 +181,16 @@ func (s *sender) logToText(record pdata.LogRecord) string {
 // logToJSON converts LogRecord to a json line, returns it and error eventually
 func (s *sender) logToJSON(record pdata.LogRecord) (string, error) {
 	data := s.filter.filterOut(record.Attributes())
-	data.orig.Upsert(logKey, record.Body())
+
+	// Only append the body when it's not empty to prevent sending 'null' log.
```
Comment (astencel-sumo) [inline pkg/exporter/sumologicexporter/sender.go:185]: How about encapsulating this `if` statement in a` function like `isEmpty`?

```
if (!isLogBodyEmpty(body)) {
  data.orig.Upsert(longKey, body)
}
```

**Review by astencel-sumo — APPROVED** [review]

File: pkg/exporter/sumologicexporter/sender.go (hunk 1 above)
Comment (pmalek-sumo) [reply pkg/exporter/sumologicexporter/sender.go:185]: Done (somewhat like proposed :) )

**Review by sumo-drosiek — APPROVED** [review]

**Review by kkujawa-sumo — APPROVED** [review]

**Pull request merged**